torch/serialization.py

```diff
@@ -465,7 +465,9 @@ def persistent_id(obj):
         # https://github.com/python/cpython/blob/master/Lib/pickle.py#L527-L537
         if torch.is_storage(obj):
             storage_type = normalize_storage_type(type(obj))
-            obj_key = str(obj._cdata)
+            if obj._cdata not in persistent_id.id_map:
+                persistent_id.id_map[obj._cdata] = str(len(persistent_id.id_map))
+            obj_key = persistent_id.id_map[obj._cdata]
```

Comment:
You can make this more concise with the [`setdefault`](https://docs.python.org/3.6/library/stdtypes.html#dict.setdefault) method:

```suggestion
            obj_key = persistent_id.id_map.setdefault(obj._cdata, str(len(persistent_id.id_map)))
```

Reply:
Nice observation, thanks!